### PR TITLE
Dev v0.6.0 add from template

### PR DESCRIPTION
**Pull request by jwildfire**
Lets users create new modules using via AddModules(). This can functionality works when the app is first created with `CreateApp()` or can be used on existing apps with a standalone call to `AddModules()`. 

When AddModules() is called, any modules listed in vModuleIDs that are not found in the specified directory are created from the specified template with the new CreateModule() function. Those newly-created module shells are then added to the app just like other pre-defined modules.

### CODE REVIEW

File: R/AddModules.R [hunk 1]
```diff
@@ -1,30 +1,51 @@
 #' @name AddModules
+#'
 #' @title AddModules
 #' @description { This function adds a module to a shiny app created by BaSS.  }
 #'
 #' @param vModuleIDs IDs of the shiny module to be added to the application. The source code for each module should be saved in strModuleDirectory with files named "mod_{strModuleID}UI.R" and mod_{strModuleID}Server.R". If desired, submodules may also be included using the following naming convention "mod_{strModuleID}_XXX.R".
+#' @param strModuleID alias for vModuleIDs. Items are merged if both are provided.
```
Comment (kwathen) [inline R/AddModules.R:7]: I am confused as to what the vModuleIDs do and strModuleID are going to do.   Not sure how I add a sub-module? Are sub-modules added if I give more than one in the vector?

**Review by kwathen — COMMENTED** [review]
I think we need to create a couple simplified functions for the user to call that I believe ultimately call AddModules.

File: inst/ProjectTemplate/_shared/app/app_ui.R [hunk 2]
```diff
@@ -48,15 +48,15 @@ app_ui <- function(){
     shinydashboard::sidebarMenu(
       id = "sidebar_tabs",
       # whisker tag to add sidebar UI
-      # {{ADD_MODULE_SIDEBAR}}
+      # {{ADD_FIRST_MODULE_SIDEBAR}}
```
[the remark this answers is not in the document]
Comment (jwildfire) [reply inst/ProjectTemplate/_shared/app/app_ui.R:51]: Yeah, that makes sense. I'll see if I can implement.

File: R/AddModules.R (hunk 1 above)
Comment (jwildfire) [reply R/AddModules.R:7]: Yeah, it's redundant. strModuleID is just added to the vModuleIDs list. Don't really need both though. Will clean it up.

**Pull request merged**